### PR TITLE
doc: Add scaling documentation

### PR DESCRIPTION
**Pull request by davidkornel**
I've created the initial scaling documentation
I included:
Why scaling is important in k8s in general?
What does scaling mean in stunner in general?
What does scaling in/out mean with stunner?
And included an example with yamls.
Please take a look at it and let me know where and how to enhance

In the meanwhile, I moved all scaling-related writings from the stunner helm documentation. Just left a link there to this doc

### CODE REVIEW

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:34]: Can we give an example? I see there's one below but we'd need it here. Maybe you could bring the example up here and only after that discuss scale-in/scale-out?

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:25]: Scaling with STUNner -> ## Scaling STUNner

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:40]: all call connections -> all connections
kick -> kick in

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:42]: I don't understand why this is an "option". Please set this as the only alternative, we support this by default. Why make people worry? Maybe after the "default" you can mention that one can switch graceful restarts off and then they will fall back to ICE restarts.

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:50]: I never know whether it's `stunner` or `stunnerd`

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:57]: We should refer the reader to a doc on how to setup HPA. Is HPA available on a default K8s install?

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:53]: As mentioned above: I'd move this example way up in the doc. We want to make this into a hands-on guide: some background and then straight to the YAMLs.

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:90]: `stunner` or `stunnerd`?

File: doc/SCALING.md
Comment (rg0now) [inline doc/SCALING.md:94]: I don't get this: do we actually generate "calls"? Or is this some hypothetical situation?

**Review by rg0now — CHANGES_REQUESTED** [review]
Thanks, I really like where this is going. Some minor changes are requested, but generally this doc fills a massive-size void in our documentation.

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:18]: included to links to pages where both the resources and the scaling types are explained

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:22]: thanks from now on i will focus on how to write bullet lists. You mentioned comma fter the items but in other docs it was semicolon so I went with it

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:28]: no, it is named after the helm chart's release name, which is I hope wont cause any trouble

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:40]: i dont get what you mean here but i admit that my sentence sounds a bit weird

Comment (davidkornel) [reply doc/SCALING.md:40]: nevermind, i get it now

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:50]: yeah u're right about this, i tried to be consistent, that when i specifially write about the application it's stunnerd and when about the pod it's stunner, 'cuz none of the kubernetes resources (pod/deployment) named stunnerd only the container and the application

File: doc/SCALING.md
Comment (levaitamas) [inline doc/SCALING.md:96]: the the -> the

As participants levate the room at the end of the meeting, the load will drop ...

**Review by levaitamas — CHANGES_REQUESTED** [review]
Thanks! I added some minor comments otherwise it looks really good.

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:14]: good idea thanks, i tried to include all the useful sources but i missed the most obvious one

File: doc/SCALING.md
Comment (davidkornel) [reply doc/SCALING.md:32]: Not sure how to change this. In this section i was focusing on scaling up. Tried to separate the two, because they follow an extremely different approach.

Comment (davidkornel) [reply doc/SCALING.md:32]: BTW when scaling up, the existing stunner infra does not change, only a new instance will come up. Wdym by my last test?

File: doc/SCALING.md
Comment (levaitamas) [inline doc/SCALING.md:77]: we don't have much to worry about -> we don't expect any breaking changes : If I get this correctly there is no need to worry about things. Why not go with the stronger and assuring claim?

File: doc/SCALING.md
Comment (levaitamas) [inline doc/SCALING.md:79]: section above -> Section 'Scaling with STUNner'  : it is recommended to avoid relative refs in docs. In case of text reorg, relative refs are a horror.

File: doc/SCALING.md
Comment (levaitamas) [inline doc/SCALING.md:91]: perhaps add a ref to back up the 10 min; for example, `[10 minutes](https://www.rfc-editor.org/rfc/rfc8656#section-3.2-3)`

**Review by levaitamas — APPROVED** [review]
Added some minor comments again. Please consider addressing them. After that I am okay with merging this PR.

**Comment by davidkornel** [comment]
@rg0now could you give another review and merge if you are satisfied? thanks

File: docs/README.md
Comment (VidarHUN) [inline docs/README.md:1]: Can you add a link to SCALING.md?

File: mkdocs.yml
Comment (VidarHUN) [inline mkdocs.yml:11]: Add SCALING.md

File: docs/README.md
Comment (davidkornel) [reply docs/README.md:1]: It was added originally but somehow i messed up, thanks

File: mkdocs.yml
Comment (davidkornel) [reply mkdocs.yml:11]: added

**Comment by rg0now** [comment]
I like this version a lot, thanks! Clean and simple. One minor note: we no longer include the Help/License footers in the docs, pls remove. Once it's done, please go ahead and squash-merge. THX!

**Comment by davidkornel** [comment]
@rg0now can you approve the changes? I cannot merge until there is a change request by you

**Pull request merged**
